src/Language/GraphQL/Draft/Printer.hs

```diff
@@ -252,8 +240,9 @@ value = \case
   VObject o   -> objectValue o
   VEnum ev    -> nameP $ unEnumValue ev
 
+-- | We use Aeson to decode string values, and therefore use Aeson to encode them back.
 stringValue :: Printer a => Text -> a
-stringValue s = mconcat [ charP '"', textP s, charP '"' ]
+stringValue s = textP $ LT.toStrict $ LT.decodeUtf8 $ J.encode s
```

Comment:
Here it f*cking is.

Reply:
Excellent catch.

Philosophically, do you think that we should use `decodeUtf8` (which throws an impure exception on invalid UTF-8 encoding) or `decodeUtf8With lenientDecode` (which will replace invalid UTF-8 characters with [the replacement character](https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character)?

From an engineering PoV I'm not opposed to the impure exception if we think this is a situation that should never happen (i.e. `Aeson.encode` should always return valid UTF-8, and if it does we want to fail hard), but I'd also suggest examining the code paths this function lies on and throwing a `catchDeep` in front of them as soon as they hit `IO`, `MonadCatch`, or `MonadUnliftIO` so we can be sure they're properly handled.

Reply:
Yeah, in this case I think the direct decode is fine, since as you point out we really rely on Aeson's output to be valid.

But, yes, I also think we should strongly simplify / harden / test our text-handling code. We do so much decoding everywhere that it's just a bug waiting to happen.